### PR TITLE
[ISSUE #375] In memory as event store

### PR DESCRIPTION
**Pull request by ruanwenjun**
Add standalone connector plugin, close issue #375 

How to use:
1. Config the connector type in eventmesh.properties
```
#connector plugin
eventMesh.connector.plugin.type=standalone
```
2. install the standalone plugin
You can add the dependency in eventmesh-runtime/build.gradle
```
implementation project(":eventmesh-connector-plugin:eventmesh-connector-standalone")
```
Or execute 
```
./gradlew clean jar dist copyConnectorPlugin
```

### CODE REVIEW

File: eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/MessageQueue.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    public MessageEntity[] items;
+
+    private int takeIndex;
+
+    private int putIndex;
+
+    private int count;
+
+    private ReentrantLock lock;
```
Comment (tydhot) [inline eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/MessageQueue.java:40]: maybe the lock is final is better in this queue.

Comment (ruanwenjun) [reply eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/MessageQueue.java:40]: Thanks, agree with you.

**Comment by xwm1992** [comment]
I try to use standalone mode to pub/sub. There is a `topic not exist exception` while client subscribe the topic, please have a look.

**Comment by ruanwenjun** [comment]
@xwm1992 Thanks, I have fixed this problem, but it seems that the consumer has another problem, I am not sure.

**Comment by xwm1992** [comment]
There is still have an exception during client subscribe to eventmesh runtime server, server didn't send the `subscribe response` cmd to the client, which cause the client operation time out.

**Comment by ruanwenjun** [comment]
@xwm1992 OK, I have fixed.

**Comment by Jackzeng1224** [comment]
@ruanwenjun   hello,When I test it will appear "SimpleSubClientImpl(SimpleSubClientImpl.java:169)-msg ignored|HEARTBEAT_RESPONSE|Package{header=Header{cmd=HEARTBEAT_RESPONSE, code=0, msg='success', seq='6462122308'), body=null}"

**Comment by ruanwenjun** [comment]
@Jackzeng1224 Thanks, but this error also appears, when I am using rocketmq.

File: eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/task/SubScribeTask.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                                        // update offset
+                                        offset.incrementAndGet();
+                                        logger.info("message commit, topic: {}, current offset:{}", topicName, offset.get());
+                                        break;
+                                    case ReconsumeLater:
+                                        // don't update offset
+                                        break;
+                                    case ManualAck:
+                                        // update offset
+                                        offset.incrementAndGet();
```
Comment (xwm1992) [inline eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/task/SubScribeTask.java:80]: Here is the ManualAck, shouldn't update the offset after commit the message, there is an acktask in runtime, only client call this task, the offset will be increment

**Review by xwm1992 — CHANGES_REQUESTED** [review]

File: eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/task/SubScribeTask.java (hunk 2 above)
Comment (ruanwenjun) [reply eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/task/SubScribeTask.java:80]: OK, please take a look.

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**